tests/conftest.py

```diff
@@ -7,4 +7,30 @@
     - https://docs.pytest.org/en/stable/writing_plugins.html
 """
 
-# import pytest
+from pathlib import Path, PurePath, PurePosixPath
+from uuid import uuid4
+
+import pytest
+
+CNFPATH = Path(__file__).resolve()
+TESTDIR = CNFPATH.parent
+DATADIR = TESTDIR / "data"
+
+UUID = str(uuid4())
+
+
+def pytest_configure():
+    pytest.UUID = UUID  # type: ignore
```

Comment:
This UUID becomes the name of the S3 bucket directory containing info about the test run, right? Is there a reason to use something like a UUID instead of a descriptive string with info about the test run (including a datetime of the run or other info about the testing)?

Reply:
Another good question! 

The UUIDs are used to create a single folder on Synapse, which will contain one folder for each test. Usually, the tests are run sequentially, and the `setUp()` and `tearDown()` methods will result in the folder being created and deleted before the next test is run. 

Originally, I was using `time.time_ns()` (current time since epoch in nanoseconds) to uniquely identify folders that I was creating. I occasionally ran into some weird issues that I presumed were related to concurrency, so I switched to a statistically unique test run ID using `uuid4()`. I haven't had the weird issues since. 

That said, I could include additional information about the test run in the name alongside the UUID to get the best of both worlds.

Reply:
+1